src/btree/bt_walk.c

```diff
@@ -402,6 +402,10 @@ __tree_walk_internal(WT_SESSION_IMPL *session, WT_REF **refp, uint64_t *walkcntp
                 couple = NULL;
                 *refp = ref;
                 WT_ASSERT(session, ref != ref_orig);
+
+                if (__wt_session_prefetch_check(session, ref))
+                    WT_ERR(__wt_btree_prefetch(session, ref));
```

Comment:
@agorrod @korteland If I understand correctly, it looks like the POC considers whether prefetching would be beneficial on a cursor/B-tree level. I don't think this is in alignment with the technical design anymore, which will turn on prefetching for all cursors if prefetching is enabled for a given session. Is this something we want to address now, or are we happy to address it in the future as this PR is quite large already?

Reply:
Thanks Alex, I'll file a new ticket and then add a FIXME comment here.

Reply:
Filed a new ticket here: [WT-11758](https://jira.mongodb.org/browse/WT-11758).